packages/@aws-cdk/aws-apigateway/lib/vpc-link.ts

```diff
@@ -31,7 +42,18 @@ export interface VpcLinkProps {
  * Define a new VPC Link
  * Specifies an API Gateway VPC link for a RestApi to access resources in an Amazon Virtual Private Cloud (VPC).
  */
-export class VpcLink extends Resource {
+export class VpcLink extends Resource implements IVpcLink {
+  /**
+   * Import a VPC Link by it's Id
+   */
+  public static fromVpcLinkId(scope: Construct, id: string, vpcLinkId: string): IVpcLink {
+    class Import extends VpcLink {
```

Comment:
Inheriting VpcLink isn't going to work. Creating a new instance of `Import` will invoke the constructor of `VpcLink` and hence going to create a new vpc link with the default values of `VpcLinkProps`.

You will have to `extend Resource` and `implement IVpcLink` just like the `VpcLink` class.

See other examples in the CDK code base.